BootloaderCommonPkg/Include/Guid/MpCpuTaskInfoHob.h

```diff
@@ -0,0 +1,50 @@
+/** @file
+  This file defines the hob structure for the AP task.
+
+  Copyright (c) 2021, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef __MP_CPU_TASK_INFO_H__
+#define __MP_CPU_TASK_INFO_H__
+
+
+///
+/// Boot loader MP CPU task Information GUID
+///
+extern EFI_GUID gLoaderMpCpuTaskInfoGuid;
+
+typedef enum {
+  EnumCpuBusy = 0,
+  EnumCpuReady,
+  EnumCpuStart,
+  EnumCpuEnd,
+} CPU_STATE;
+
+typedef UINT32 (*CPU_TASK_PROC)          (UINT32 Arg);
+
+typedef struct {
+  // Refer CPU_STATE
+  UINT32           State;
+
+  // CPU task function
+  CPU_TASK_PROC    CProcedure;

```

Comment:
Better to use a  UINT64 instead of pointer since the size will be different between 32bit and 64bit environment.

Reply:
This is the flag to request to start running the CProcedure.   The old code use CProcedure itself as a flag to trigger.  It is OK to do that as well if we can ensure the write is atomic.   Using a separate byte filed might be safer.

Reply:
Based on current impl, it should be ok as long as we always set State to EnumCpuStart at the end.